node/src/chain_spec.rs

```diff
@@ -63,17 +92,24 @@ pub struct Extensions {
 /// Specialized `ChainSpec`. This is a specialization of the general Substrate ChainSpec type.
 pub type ChainSpec = sc_service::GenericChainSpec<GenesisConfig, Extensions>;
 
-fn session_keys(
-	grandpa: GrandpaId,
-	babe: BabeId,
-	im_online: ImOnlineId,
-	authority_discovery: AuthorityDiscoveryId,
-) -> SessionKeys {
-	SessionKeys {
-		grandpa,
-		babe,
-		im_online,
-		authority_discovery,
+#[derive(Clone)]
+pub struct InitialAuthority {
+	pub stash: AccountId,
+	pub stash_balance: Balance,
+	pub controller: AccountId,
+	pub controller_balance: Balance,
+	pub session_keys: SessionKeys,
+}
+
+impl InitialAuthority {
+	fn new(stash: AccountId, controller: AccountId, grandpa: GrandpaId) -> Self {
+		Self {
+			stash,
+			stash_balance: 1_000_000 * AVL,
```

Comment:
Is there any value in parametarizing the stash and controller balance?

Reply:
I think we do not need it for TestNet, but It could be a requirement on MainNet.